netbout-web/src/main/java/com/netbout/dynamo/DyMessages.java

```diff
@@ -193,23 +195,27 @@ public void post(final String text) throws IOException {
 
     @Override
     public long unread() throws IOException {
-        final Item item = this.region.table(DyFriends.TBL)
-            .frame()
-            .through(
-                new QueryValve()
-                    .withLimit(1)
-                    .withAttributesToGet(DyFriends.ATTR_UNREAD)
-            )
-            .where(DyFriends.HASH, Conditions.equalTo(this.bout))
-            .where(DyFriends.RANGE, Conditions.equalTo(this.self))
-            .iterator().next();
-        final long unread;
-        if (item.has(DyFriends.ATTR_UNREAD)) {
-            unread = Long.parseLong(item.get(DyFriends.ATTR_UNREAD).getN());
-        } else {
-            unread = 0L;
+        try {
+            final Item item = this.region.table(DyFriends.TBL)
+                .frame()
+                .through(
+                    new QueryValve()
+                        .withLimit(1)
+                        .withAttributesToGet(DyFriends.ATTR_UNREAD)
+                )
+                .where(DyFriends.HASH, Conditions.equalTo(this.bout))
+                .where(DyFriends.RANGE, Conditions.equalTo(this.self))
+                .iterator().next();
```

Comment:
@2686747 the problem here is not that `next()` throws an exception, the problem is calling it when there are no more elements in the iterator.
Every iterator defines `hasNext()`, you should use that to avoid making the erroneous `next()` call instead of making it and suppressing the error.